### ISSUE TITLE
API docs for .NET SDK

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/docs#2137]
Add .NET to the API docs at https://www.pulumi.com/docs/reference/pkg/

**Comment by lukehoban** [comment]
@mikhailshilkov @praneetloke Our plan of record right now is that we will have resource-level docs as our primary entrypoint from the docs site - but that these will link into language-specific docs for details of individual language projections.  That means we will still need .NET API docs to be available.  Is the plan to publish and host those on `pulumi.com/docs` (as we currently do for TypeScript and Python) or to host somewhere externally (as we do for Go)?

**Comment by praneetloke** [comment]
My understanding is these will need to be hosted by us. I haven't looked into this much yet, but from what I know there isn't a central place for dotnet docs similar to pkg.go.dev.

**Comment by lukehoban** [comment]
I think @mikhailshilkov had a WIP on this - so I'll assign it to him.  I believe he was just blocked on: 
1. Us deciding the plan for resoruce-model and/or API model docs (we've now decided on "both")
2. Issues with breaking Pulumify when adding thousands of new files (we should not block on this - if Pulumify can't yet handle the content we need to host on this site - we should stop using Pulumify until it can be fixed to support it)

**Comment by praneetloke** [comment]
> Issues with breaking Pulumify when adding thousands of new files (we should not block on this - if Pulumify can't yet handle the content we need to host on this site - we should stop using Pulumify until it can be fixed to support it)

Yes, this is a problem without adding the .NET docs as well. Just with the new resource-level docs, Pulumify does not work in its current form. See this [comment](https://github.com/pulumi/docs/pull/2551#issuecomment-592175142) in this prototype PR opened by Justin.

**Comment by mikhailshilkov** [comment]
Another challenge is related to styling these pages. How much time do we want to spend on making them look and feel as the rest of the site? If we want ~100% flexibility, it might make sense to write our own tool akin to ones for TS/Python instead of trying to fit the default HTML.

Is there somebody with better HTML/CSS skills than me that I could discuss this with?

**Comment by praneetloke** [comment]
@mikhailshilkov let me connect with you offline.

**Comment by mikhailshilkov** [comment]
An early PR: https://github.com/pulumi/docs/pull/2172
that shows the preview:
- http://content-bucket-a3491d9.s3-website-us-east-1.amazonaws.com/docs/reference/pkg/dotnet/Pulumi/Pulumi.StackReference.html
- http://content-bucket-a3491d9.s3-website-us-east-1.amazonaws.com/docs/reference/pkg/dotnet/Pulumi.Azure/Pulumi.Azure.AppService.FunctionApp.html

**Comment by mikhailshilkov** [comment]
EventStore is an example of using DocFX:
- Site: https://eventstore.com/docs/
- Repo: https://github.com/EventStore/documentation
- Template: https://github.com/EventStore/es-docs-template/

**Comment by praneetloke** [comment]
We had an offline sync-up for this.

* We will generate the docs using docfx as shown in Mikhail's WIP PR #2172.
* The docs would be housed under the `static/` directory, so that it doesn't conflict with Hugo's processing.
* As for the layout of the docs pages, @cnunciato has agreed to step-in to help make the page look/feel like it is part of the main site. Christian will first need the docs generated, so that he can take a look at the layouts and see what can be done with them, so this work item will be an M33 thing. I'll file a separate issue for this.

**Comment by mikhailshilkov** [comment]
The initial prototype is now available in the updated https://github.com/pulumi/docs/pull/2172
